client/core/trade.go

```diff
@@ -1470,6 +1483,17 @@ func (c *Core) finalizeSwapAction(t *trackedTrade, match *matchTracker, coinID,
 		errs.add("error storing swap details in database for match %s, coin %s: %v",
 			match.id, coinIDString(t.wallets.fromAsset.ID, coinID), err)
 	}
+
+	// With updated swap data, attempt match status resolution.
+	if needsResolution {
```

Comment:
Now making sure match status and proof data updates happen before firing off the resolution goroutine.